src/client/mangrove.ts

```diff
@@ -12,20 +12,23 @@ import {Observable} from './utils/observable'
 
 export const documentSelector: DocumentSelector = [
 	{language: 'mangrove', scheme: 'file'},
-	{language: 'mangrove', scheme: 'untitled'},
+	{language: 'mangrove', scheme: 'untitled'}
 ]
 
 export async function createLanguageClient(folder: WorkspaceFolder): Promise<LanguageClient>
 {
 	const clientOptions: LanguageClientOptions =
 	{
-		documentSelector: documentSelector,
+		documentSelector,
 		diagnosticCollectionName: 'mangrove',
 		initializationOptions: workspace.getConfiguration('mangrove'),
 		workspaceFolder: folder
 	}
 
-	const languageServer = extensionContext.asAbsolutePath(path.join('build', 'server', 'server.js'))
+	const languageServer = extensionContext?.asAbsolutePath(path.join('build', 'server', 'server.js')) ?? ''
```

Comment:
This nullish coalescing to empty string feels unnecessary as the value will be falsy anyways